.github/workflows/check-untracked-repos.yml

```diff
@@ -0,0 +1,42 @@
+name: Check Untracked Repositories
+
+on:
+  schedule:
+    - cron: '0 */6 * * *'
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+  check-untracked-repos:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Install Rust Stable
+        env:
+          RUST_VERSION: "1.91.1"
+        run: |
+          rustc -vV
+          rustup toolchain install $RUST_VERSION
+          rustup default $RUST_VERSION
+          rustup component add rustfmt clippy
+          rustc -vV
+
+      - name: Cache Cargo dependencies
+        uses: actions/cache@v4
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: Check for untracked repositories
+        run: cargo run --bin rust-team ci check-untracked-repos
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
this is the token with write access.
Instead we want to use the token with read only access, which is the one we use in the dry run. So I'm afraid we need to do something like this:

https://github.com/rust-lang/team/blob/bbdb6ce9b660bff3b7c535f2ed6358d80ddd9cae/.github/workflows/dry-run.yml#L120

Is there a way to simplify the workflow without having to copy-paste? Not sure.

Reply:
I *think* that secrets can't be passed between jobs, so a reusable job probably wouldn't work. But a reusable workflow might, if we parametrized what gets executed in it.